ons-spark/spark-analysis/interpolation.md

```diff
@@ -0,0 +1,502 @@
+   area_code period   count impute_flag timestamp_with_nulls timestamp_all
+   <chr>     <chr>    <dbl>       <dbl>                <dbl>         <dbl>
+ 1 B         20210101    NA           1                   NA    1609459200
+ 2 B         20210111    85           0           1610323200    1610323200
+ 3 B         20210119    82           0           1611014400    1611014400
+ 4 B         20210131    75           0           1612051200    1612051200
+ 5 B         20210210    NA           1                   NA    1612915200
+ 6 B         20210215    85           0           1613347200    1613347200
+ 7 B         20210220    NA           1                   NA    1613779200
+ 8 B         20210225    75           0           1614211200    1614211200
+ 9 A         20210101   100           0           1609459200    1609459200
+10 A         20210106    NA           1                   NA    1609891200
+11 A         20210111    NA           1                   NA    1610323200
+12 A         20210116   130           0           1610755200    1610755200
+13 A         20210121   100           0           1611187200    1611187200
+14 A         20210126   120           0           1611619200    1611619200
+15 A         20210131    NA           1                   NA    1612051200
+16 A         20210205   130           0           1612483200    1612483200
+   count_ff count_bf  period_ff  period_bf
+      <dbl>    <dbl>      <dbl>      <dbl>
+ 1       NA       85         NA 1610323200
+ 2       85       85 1610323200 1610323200
+ 3       82       82 1611014400 1611014400
+ 4       75       75 1612051200 1612051200
+ 5       75       85 1612051200 1613347200
+ 6       85       85 1613347200 1613347200
+ 7       85       75 1613347200 1614211200
+ 8       75       75 1614211200 1614211200
+ 9      100      100 1609459200 1609459200
+10      100      130 1609459200 1610755200
+11      100      130 1609459200 1610755200
+12      130      130 1610755200 1610755200
+13      100      100 1611187200 1611187200
+14      120      120 1611619200 1611619200
+15      120      130 1611619200 1612483200
+16      130      130 1612483200 1612483200
+```
+
+We're now ready to calculate the gradient between the known points and interpolate. 
+
+````{tabs}
+```{code-tab} py
+# Create a gradient column
+df = df.withColumn("gradient",
+                   F.when(F.col("impute_flag")==1,
+                          (F.col("count_bf") - F.col("count_ff")) /
+                          (F.col("period_bf") - F.col("period_ff")))
+                   .otherwise(0)
+                  )
+
+# Create the imputed values
+df = df.withColumn("count_final",
+                  F.when(F.col("impute_flag")==1,
+                        F.col("count_ff") +
+                        F.col("gradient") *
+                        (F.col("timestamp_all") - F.col("period_ff"))
+                        )
+                   .otherwise(F.col("count"))
+                  )
+```
+```
```

Comment:
Missing "{code-tab} r R" When converting to R script this block was missed causing an error, with this corrected the script runs correctly